qutebrowser/browser/hints.py

```diff
@@ -130,7 +137,22 @@ def _move_to_elem(self):
     def cleanup(self):
         """Clean up this element and hide it."""
         self.hide()
-        self.deleteLater()
+        self.setParent(None)
+        self._context.tab.contents_size_changed.disconnect(self._move_to_elem)
+        HintLabel._object_pool.append(self)
+
+    @staticmethod
+    def create(elem, context):
+        """Returns a HintLabel object with the parameters (elem, context).
+        Try to reuse an object in the object pool if possible.
+        """
+
+        try:
+            self: HintLabel = HintLabel._object_pool.pop()
```

Comment:
Type hinting makes the code pass the pylint tests, but fails the tests with python 3.5.